spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -410,20 +417,43 @@ def __init__(
                 for s_id in self._unit_ids:
                     self._spiketrains.append(res[(clu == s_id + 1).nonzero()])  # from 2,...,clu[0]-1
 
+        if spkfile_path is not None and Path(spkfile_path).is_file():
+            n_bits = int(xml_root.find('acquisitionSystem').find('nBits').text)
+            dtype = f"int{n_bits}"
+            n_samples = int(xml_root.find('neuroscope').find('spikes').find('nSamples').text)
+            wf = np.memmap(spkfile_path, dtype=dtype)
+            n_channels = int(wf.size / (n_spikes * n_samples))
+            wf = np.moveaxis(wf.reshape(n_spikes, n_samples, n_channels), 1, -1)
```

Comment:
```suggestion
            wf = np.memmap(spkfile_path, dtype=dtype, shape=(n_spikes, n_samples, -1))
            wf = np.moveaxis(wf, 1, -1)
```

see https://numpy.org/doc/stable/reference/generated/numpy.memmap.shape.html

Reply:
I almost always use the shape argument on memmap instantiation, but in this case it would be a bit messier then usual for a couple reasons.

1) The memmap `shape` argument doesn't automatically fill dimension values via `-1`, only reshape handles that. 
2) As seen in the code, the `n_spikes` can be easily inferred from the `.clu` file, the `n_samples` can be easily found in the `.xml`; the `n_channels` information is technically within the `.xml` file as well, but would be just a tad more complicated to extract (estimating an additional 2-4 lines of code to do it safely) because these lines of code support both single shank and multi shank channel groups, and the only information we have about which channel group it is comes from. Would also have to debug it on single-pair res/clu .xml, which I haven't done in a while.
3) Even beyond the extra code (2) would requires, there's actually practical reasons to use `-1` instead of reading the true channel groups from the `.xml` - on occasion, I notice that certain channels are outright removed from the `.spk` file, likely because they were considered "bad" at some point in post-processing. As a result, there's a mismatch between the groups reported in the `.xml` and the actual amount of data in the `.spk`.

That said, I have condensed these three lines into one single call, if that looks better.